docs/storage-config.md

```diff
@@ -0,0 +1,204 @@
+to be enabled. If the PVC references a networked filesystem that can be mounted
+on multiple hosts, then VMs can migrate with no need to copy emphemeral disks.
+If multiple volume mounts are made available to the libvirtd POD, annotations
+would need to be used to tag them for different usage types. This might be useful
+if a cluster needs to provide ephemeral storage with differing performance grades
+or costs (eg one volume provides cheap & slow rotating rust, while another volume
+provides expensive & fast sold state).  The VM spec disk configuration would be
+able to optionally include a reference to the usage type it desires, to allow
+virt-handler to map the disk to the particular volume mount.
+
+Thus the libvirtd container may contain
+
+  annotations:
+    kubevirt.io/storage.usage-type.fast: /storage/ssd
+    kubevirt.io/storage.usage-type.slow: /storage/rust
+  volumeMounts:
+    - mountPath: /storage/ssd
+      name: storage-ssd
+    - mountPath: /storage/rust
+      name: storage-rust
+
+NB The DaemonSet resource does not provide a way to dynamically assign
+different PVCs to each libvirtd POD it creates. Thus in the short term, to make
+use of this feature would require the cloud admin to manually create libvirtd
+POD specs for each compute node, instead of using a DeamonSet resource. It is
+anticipated that either DaemonSet would be enhanced eventually, or replaced by
+a more advanced resource that can dynamically assigns PVCs to each POD.
+
+When configuring the VM spec disk
+
+    disks:
+      - source:
+          pvc:
+	    claimname: some-pvc-name
+	access-mode: read-only
+	copy:
+	  mode: copy-on-write
+          usage-type: fast
+
+The 'usage-type' attribute refers to the 'kubevirt.io/sotrage.usage-type'
+annotations provided by the libvirtd POD. Either all libvirtd PODs would need
+to support the same annotations, or the scheduler will need to be able to only
+place VMs on the nodes where the libvirtd POD has the neccessary annotations.
+
+Each of the volume mounts provided for emphemeral storage would need to have a
+libvirt storage pool associated with them. To create the ephemeral disks, the
+virt-handler would thus use the storage pool APIs in libvirtd to create disks
+whose backing store refers to the master PVC.
+
+
+## Populating PVCs
+
+The virtual machine startup code path should not care how the PVCs associated
+with the disks are populated. It is assumed that the PVCs have been setup ahead
+of time, either by the cloud administrator, or by the tenant user. There are a
+number of possible approaches that could be used to populate these PVCs, the
+choice of which is opaque to virt-handler, where merely needs the PVC name.
+
+This doc will outline some high level concepts, but detailed technical designs
+for them will be left for separate documents to define.
+
+### HTTP upload
+
+A "HTTP upload" POD could be created that references the PVC that needs to
```

Comment:
Presumably we would only want this pod to run temporarily--long enough for the user to upload their intended image, then it would exit. It might be worth mentioning that.